### DISCUSSION TITLE
Pull request failures

### DISCUSSION THREAD

**Discussion by derekwyatt** [derekwyatt/vim-scala#165] [General]
Hey @ches,

I see we have a bit of a problem getting a PR working for mainline vim. Obviously I haven't been involved in this in quite some time as I sadly don't code in Scala anymore and am not actively testing any of this work.

Ultimately, it would be best to find a new home for this with someone that is lucky enough to be coding in Scala regularly, but for the present moment, I think we might want to clear up all of the outstanding PRs and get them properly into Vim mainline.

There are failures on the PR testing that don't _appear_ to have anything to do with the PR itself, but it's still on us to deal with it, I think.  How do you feel about pulling it back, and replacing it with a PR that clears up all of the PRs against our branch?  We'll get this long-lived backlog out of the way, get it passing tests, and we'll have a clean slate from which to discuss transitioning the repo entirely.

I'd be very happy to get on Zoom and just work it through until we've got it sorted.

**Comment by ches** [comment]
Hey Derek!

Thanks for starting the discussion, and sorry to hear you're not working in Scala these days 😞

I am, but I haven't been making much time for open source maintenance lately, and honestly I'd forgotten for a long time that I had merge rights here… I got motivated one weekend day a couple of weeks ago when we were pinged on one of the PRs, I went through several and verified + merged a few, and that's when I sent vim/vim#8748 with everything I saw that hadn't gone to mainline Vim yet.

I will try to get through most of the remaining outstanding ones next weekend. I don't mind trying to stay active here with reviewing and merging—now that I've remembered I'm able—and occasionally sending changes to Bram.

Maybe we can create a vim-scala organization to hold it, I can also help vetting other people as maintainers as time goes on.

I'd like to get some kind of CI going, especially now that GitHub Actions makes it easier to manage in one place. Not sure if the Ruby Vimspec stuff that's here can be easily revived or if there are better options now, but it's worth a try. Manually testing things definitely won't inspire me to be a good maintainer…

Anyway, I think the current PR to Vim just needs a sign-off, but I'd be glad to chat if you want to work through any PRs or administrative stuff.

**Comment by derekwyatt** [comment]
I've thrown a 👍 at Bram so that should unblock the merge.

Putting some CI in place is a fantastic idea. We could hook that up and then make sure we've got a decent number of tests for it to make us comfortable. Do you happen to know if there's a Vim repo out there that we can use as an excellent example of this sort of thing?  I can't imaging how one might test the syntax rendering. I've always just eyeballed it, and that's not exactly the most ideal :)

In terms of the next immediate action, that seems like the right one to me. I'll make an issue for it, and we'll see where it goes.

Thx man :)
